### ISSUE TITLE
TensorFlow reverting to GPU on a CPU machine on most recent nightly build

### ISSUE THREAD

**Issue by helpimalive** [tensorflow/tensorflow#9302]
I'm running tensorflow on windows. I continue to produce these errors around CUDA libraries and cuDNNs ,but I don't have a GPU and didn't install a GPU version of TF. 

I installed the latest nightly build of TF from jenkins (apr 17) but when that one failed on me I tried to pip3 uninstall and revert to [this one](https://ci.tensorflow.org/view/Nightly/job/nightly-win/DEVICE=cpu,OS=windows/134/) to no success.

I was up and running last night thanks to a solution on github from Derek, but cannot stop reproducing this CUDA error after playing around with the path variables in pycharm.

- Error is reproduced on both custom code and standard wide_n_deep TensorFlow examples
- Windows 10 64; Python 35
- TF installed from nightly build
- TF versions 1.1.0 and 1.0.1
- I have no GPU
- C:\Users\me>C:\Users\me\AppData\Local\Programs\Python\Python35\python.exe C:\Users\me\Desktop\wide_n_deep --model_type ="wide_n_deep"

### Source Code / Logs
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:126] Couldn't open CUDA library cublas64_80.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_blas.cc:2294] Unable to load cuBLAS DSO.
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:126] Couldn't open CUDA library cudnn64_5.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_dnn.cc:3517] Unable to load cuDNN DSO
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:126] Couldn't open CUDA library cufft64_80.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_fft.cc:344] Unable to load cuFFT DSO.
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:126] Couldn't open CUDA library nvcuda.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_diagnostics.cc:165] hostname: DESKTOP-HMIARON
Traceback (most recent call last):
  File "C:\Users\Larriva\AppData\Roaming\Python\Python35\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Users\Larriva\AppData\Local\Programs\Python\Python35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 906, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: A dynamic link library (DLL) initialization routine failed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Larriva\AppData\Roaming\Python\Python35\site-packages\tensorflow\python\__init__.py", line 66, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Users\Larriva\AppData\Roaming\Python\Python35\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\Users\Larriva\AppData\Roaming\Python\Python35\site-packages\tensorflow\python\pywrap_tensorflow.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow')
  File "C:\Users\Larriva\AppData\Local\Programs\Python\Python35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Larriva\Desktop\wnd2.py", line 15, in <module>
    import tensorflow as tf
  File "C:\Users\Larriva\AppData\Roaming\Python\Python35\site-packages\tensorflow\__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "C:\Users\Larriva\AppData\Roaming\Python\Python35\site-packages\tensorflow\python\__init__.py", line 72, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\Users\Larriva\AppData\Roaming\Python\Python35\site-packages\tensorflow\python\pywrap_tensorflow.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Users\Larriva\AppData\Local\Programs\Python\Python35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 666, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 577, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 906, in create_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
ImportError: DLL load failed: A dynamic link library (DLL) initialization routine failed.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Larriva\AppData\Roaming\Python\Python35\site-packages\tensorflow\python\__init__.py", line 66, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Users\Larriva\AppData\Roaming\Python\Python35\site-packages\tensorflow\python\pywrap_tensorflow.py", line 21, in <module>
    _pywrap_tensorflow = swig_import_helper()
  File "C:\Users\Larriva\AppData\Roaming\Python\Python35\site-packages\tensorflow\python\pywrap_tensorflow.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow')
  File "C:\Users\Larriva\AppData\Local\Programs\Python\Python35\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: No module named '_pywrap_tensorflow'

Failed to load the native TensorFlow runtime.

See https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/get_started/os_setup.md#import_error

for some common reasons and solutions.  Include the entire stack trace
above this error message when asking for help.

**Comment by drpngx** [comment]
Did you try to reboot? Is that with the new Windows 10 Creator update? Do you have up to date nvidia drivers?

**Comment by helpimalive** [comment]
Thank you for your reply. I tried a reboot.
I've not done any windows updates in the time between having the working code and non-working code.
I don't believe I have any NVIDIA hardware or software on my computer. Intel runs the graphics.

**Comment by drpngx** [comment]
Are these in your path?

* `cublas64_80.dll`
* `cudnn64_5.dll`
* `cufft64_80.dll`
* `nvcuda.dll`

If not, try to locate them and add them into your path. You can use the windows dependency walker to check if you need to add more dependencies.

/CC: @mrry

**Comment by helpimalive** [comment]
those dlls don't exist on my system anywhere, so was unable to add them to PATH.
I'm downloading cuda_8.0.61_win10 in hopes of grabbing those dlls...
https://developer.nvidia.com/cuda-downloads
installer recommended a different specific version of visual studio
https://www.visualstudio.com/thank-you-downloading-visual-studio/?sku=Community&rel=15
cuda install failed because the above version of VS is 17 and cuda wants <=2015.
Found a version of 2015 
http://download.cnet.com/Visual-Studio-Community-2015/3055-2212_4-76440611.html?tag=pdl-redir
Reinstalled the CUDA package
Came closer and got a few more packages successfully found, but issue did not resolve...

`I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:135] successfully opened CUDA library cublas64_80.dll locally
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:126] Couldn't open CUDA library cudnn64_5.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_dnn.cc:3517] Unable to load cuDNN DSO
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:135] successfully opened CUDA library cufft64_80.dll locally
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:126] Couldn't open CUDA library nvcuda.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_diagnostics.cc:165] hostname: DESKTOP-HMIARON`

**Comment by drpngx** [comment]
Yes, plesae install the rest of the DLLs.

**Comment by helpimalive** [comment]
Do you know where I can download?
They didn't seem to come with the CUDA package.

**Comment by drpngx** [comment]
Please follow the resolution there. 

https://github.com/tensorflow/tensorflow/issues/5968

**Comment by helpimalive** [comment]
I appreciate your help but i'm still not running. 
None of the CUDA packages seemed to contain nvcuda.dll so I grabbed the ala carte  dll off a site and dropped it into the same bin as the other referenced dlls, re-added it to PATH and still am receiving the message:
```

I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:135] successfully opened CUDA library cublas64_80.dll locally
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:135] successfully opened CUDA library cudnn64_5.dll locally
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:135] successfully opened CUDA library cufft64_80.dll locally
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:126] Couldn't open CUDA library nvcuda.dll
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\cuda\cuda_diagnostics.cc:165] hostname: ê≡° 
```

**Comment by helpimalive** [comment]
Solved it. I believe it hints at my original issue: something somewhere in the process made TF look for GPU components despite my not having a GPU.
I uninstalled everything (python, tensorflow, dumped the cached files in the pip bin) then got a nightly build of TF from Jenkins, reinstalled Python, deleted my VS17 (because the pywrap requries the15 build) and got a successful load.
Thank you again for your help.
